MultiIRIS follow up

YoshikiMas opened the pull request:
This is a follow up for [my previous PR for MultiIRIS](https://github.com/espnet/espnet/pull/4706). I add `run_multiiris.sh` in `egs2/chime4/enh_asr1/local`.
This PR is not ready for merge yet. The training with it is ongoing, and I will add usage on README and the pre-trained model.

Emrys365 commented on egs2/chime4/asr1/local/data.sh:
```diff
@@ -85,12 +85,6 @@ if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
     local/real_enhan_chime4_data_prep.sh beamformit_5mics ${PWD}/enhan/beamformit_5mics
     local/simu_enhan_chime4_data_prep.sh beamformit_5mics ${PWD}/enhan/beamformit_5mics
 
-    # prepare data for 6ch track:
-    #  (1) {tr05,dt05,et05}_simu_isolated_6ch_track
-    local/simu_ext_chime4_data_prep.sh --track 6 isolated_6ch_track ${PWD}/local/nn-gev/data/audio/16kHz
-    #  (2) {tr05,dt05,et05}_real_isolated_6ch_track
-    local/real_ext_chime4_data_prep.sh --track 6 isolated_6ch_track ${CHIME4}/data/audio/16kHz/isolated_6ch_track
-
```
What is the intension here?

YoshikiMas replied:
In my last pull request, I added these lines for [chime4_asr_data.sh](https://github.com/espnet/espnet/blob/master/egs2/chime4/enh_asr1/local/chime4_asr_data.sh) in enh_asr1.
But, now I notice the same scripts are executed in [chime4_enh_data.sh](https://github.com/espnet/espnet/blob/master/egs2/chime4/enh_asr1/local/chime4_enh_data.sh) that is a symbolic link to `data.sh` in enh1.
So, they are redundnat and should be removed because `simu_ext_chime4_data_prep.sh` and `simu_ext_chime4_data_prep.sh` are not in `chime4/asr1/local`.
I'm really sorry for missing this point.

Emrys365 replied:
Oh I see. Thanks for the clarification.

Emrys365 commented on egs2/chime4/enh_asr1/local/run_asr_pretraining.sh:
```diff
@@ -0,0 +1,39 @@
+set -u
+set -o pipefail
+
+train_set=
+valid_set=
+test_sets=
+asr_config=
+inference_config=
+lm_config=
+use_word_lm=
+word_vocab_size=
```
How should we fill these values? Could you add the default values if possible?

Emrys365 commented on egs2/chime4/enh_asr1/local/run_asr_pretraining.sh:
```diff
@@ -0,0 +1,39 @@
+cd ../../asr1
+
+./asr.sh                                   \
+    --stage 1 \
+    --stop_stage 11 \
```
We often let the user to specify the stages by adding the argument `"$@"` at the end.

YoshikiMas replied:
I expect that users do not call `run_asr_pretraining.sh` directly (like other files in local). I assume that users will execute `run_multiiris.sh` that calls `run_enh_pretraining.sh` , `run_asr_pretraining.sh`, and `run_enh_asr_finetuning.sh`.
So, the default values are summarized in `run_multiiris.sh`.

If the name `run_xxx.sh` is misleading, I would like to rename them.

Emrys365 commented on egs2/chime4/enh_asr1/local/run_asr_pretraining.sh:
```diff
@@ -0,0 +1,39 @@
+    --bpe_train_text "data/${train_set}/text" \
+    --lm_train_text "data/${train_set}/text data/local/other_text/text"
+
+cd $dir
```
I think this line can be removed.

Emrys365 commented on egs2/chime4/enh_asr1/local/run_enh_asr_finetuning.sh:
```diff
@@ -0,0 +1,50 @@
+set -u
+set -o pipefail
+
+stage=1
+stop_stage=15
+train_set=
+valid_set=
+test_sets=
+enh_asr_config=
+inference_config=
+lm_config=
+use_word_lm=
+word_vocab_size=
+extra_annotations=
+ref_channel=
+lm_exp=
+inference_enh_asr_model=
```
ditto

Emrys365 commented on egs2/chime4/enh_asr1/local/run_enh_pretraining.sh:
```diff
@@ -0,0 +1,34 @@
+set -u
+set -o pipefail
+
+train_set=
+valid_set=
+test_sets=
+enh_config=
+extra_annotations=
+ref_channel=
```
ditto

sw005320 commented:
@YoshikiMas, `README.md` is not easy to understand
Can you refine it?
You can also check `README.md` in the other recipes
https://github.com/espnet/espnet/blob/master/egs2/librispeech/asr1/README.md
https://github.com/espnet/espnet/blob/master/egs2/aishell/asr1/README.md

sw005320 commented:
Thx!

YoshikiMas commented:
> @YoshikiMas, `README.md` is not easy to understand Can you refine it? You can also check `README.md` in the other recipes https://github.com/espnet/espnet/blob/master/egs2/librispeech/asr1/README.md https://github.com/espnet/espnet/blob/master/egs2/aishell/asr1/README.md

@sw005320, thank you for pointing out and sharing the references. I changed the order and added the link to each config.
I would like to keep some kinds of notes like [https://github.com/espnet/espnet/tree/master/egs2/librimix/enh_diar1](https://github.com/espnet/espnet/tree/master/egs2/librimix/enh_diar1) so that users can easily find `run_multiiris.sh` in `local`. Could you give detailed comments to improve this point?

sw005320 commented:
> > @YoshikiMas, `README.md` is not easy to understand Can you refine it? You can also check `README.md` in the other recipes https://github.com/espnet/espnet/blob/master/egs2/librispeech/asr1/README.md https://github.com/espnet/espnet/blob/master/egs2/aishell/asr1/README.md
> 
> @sw005320, thank you for pointing out and sharing the references. I changed the order and added the link to each config. I would like to keep some kinds of notes like https://github.com/espnet/espnet/tree/master/egs2/librimix/enh_diar1 so that users can easily find `run_multiiris.sh` in `local`. Could you give detailed comments to improve this point?

Can you just rename @simpleoier's model (https://github.com/espnet/espnet/blob/master/egs2/chime4/enh_asr1/README.md#results) to be aligned with your modification so that people can understand the difference of these two models easily?
